tests/trainer/test_supporters.py

```diff
@@ -182,3 +191,73 @@ def test_combined_loader_sequence_max_size_cycle():
         assert len(item) == 2
 
     assert idx == len(combined_loader) - 1
+
+
+class IndexedRandomDataset(RandomDataset):
+
+    def __getitem__(self, index):
+        return {"index": index, "batch": self.data[index]}
+
+
+class TestModel(BoringModel):
+
+    def __init__(self, numbers_test_dataloaders, save_preds_on_dl_idx):
+        super().__init__()
+        self._numbers_test_dataloaders = numbers_test_dataloaders
+        self._save_preds_on_dl_idx = save_preds_on_dl_idx
+
+    def test_step(self, batch, batch_idx, dataloader_idx=None):
+        indexes, x = batch["index"], batch["batch"]
+        output = self.layer(x)
+        loss = self.loss(batch, output)
+
+        if dataloader_idx is not None and dataloader_idx == 1 and not self._save_preds_on_dl_idx:
+            return {"y": loss}
+
+        predictions = []
+        for idx in range(len(indexes)):
+            id = indexes[idx]
+            prediction = output[idx]
+            predictions.append({"id": id, "prediction":prediction})
+
+        self.add_predictions(predictions)
```

Comment:
can you create fake predictions and check if the value is the expected in the test below? not just the length

Reply:
Good call !